### PR TITLE
PEP 658: Static Distribution Metadata in the Simple Repository API

### PR DESCRIPTION
**Pull request by uranusjr**
Spawned from pypa/warehouse#8254. Missing fields to be filled later after a number is assigned.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
You can take PEP 658; #1944 will need to be renumbered anyway.

**Comment by pfmoore** [comment]
This looks broadly good to me. I spotted a couple of typos, I'll do a proper review when I have more time.

File: pep-0658.rst [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+Content-Type: text/x-rst
+Created: 10-May-2021
+Post-History: 10-May-2021
+Resolution:
+
+
+Abstract
+========
+
+This PEP proposes adding anchor tag to expose the ``METADATA`` file
```
Comment (JelleZijlstra) [inline pep-0658.rst:18]:
```suggestion
This PEP proposes adding an anchor tag to expose the ``METADATA`` file
```

File: pep-0658.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+
+Motivation
+==========
+
+Package management workflows made popular by recent tooling increase
+the need to inspect distribution metadata without intending to install
+the distribution, and download multiple distributions of a project to
+choose from based on their metadata. This means they end up discarding
+much downloaded data, which is inefficient and results in bad user
```
Comment (JelleZijlstra) [inline pep-0658.rst:31]:
```suggestion
much downloaded data, which is inefficient and results in a bad user
```

**Comment by uranusjr** [comment]
Thanks, I’ve assigned the number and added the newly-created Discourse link to the document. I admit I didn’t re-read the document myself after writing it 😓

File: pep-0658.rst [hunk 3]
```diff
@@ -0,0 +1,178 @@
+PEP: 658
+Title: Static Distribution Metadata in the Simple Repository API
+Author: Tzu-ping Chung <uranusjr@gmail.com>
+Sponsor:
```
Comment (pradyunsg) [inline pep-0658.rst:4]: @pfmoore here, and in code owners?

Comment (uranusjr) [reply pep-0658.rst:4]: I’m not sure whether Paul or Donald is better to put in which fields TBH

Comment (pfmoore) [reply pep-0658.rst:4]: As this is package index stuff, @dstufft is PEP-delegate by default. He can get someone else to handle it if he prefers, and I'd be willing to do it if he wants to pass the job on.

I'd rather not be sponsor, TBH. With the new CODEOWNERS workflow, it looks like that means I'd now be responsible for merging any PRs to the PEP and I don't really have the time to handle that right now...

Comment (JelleZijlstra) [reply pep-0658.rst:4]: > With the new CODEOWNERS workflow, it looks like that means I'd now be responsible for merging any PRs to the PEP

I'm happy to remain as chief button-pusher to merge uncontroversial PRs :). I think the main point of the CODEOWNERS change is to make it easier for sponsors to stay up to date on changes to their PEPs, but they don't necessarily have to take care of all changes.

Comment (pradyunsg) [reply pep-0658.rst:4]: Donald is gonna be the delegate IIUC, based on the SC's standing delegation for PyPI.

Sponsor is... well, whoever ends up sponsoring this.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by uranusjr** [comment]
The sponsor has been set to @brettcannon ([ref](https://discuss.python.org/t/8651/18)) and the PEP delegate @dstufft ([as the standing delegate](https://github.com/python/steering-council/blob/main/process/standing-delegations.md#pypa-delegations)).

I also made some changes to the proposal according to early feedback.

**Pull request merged**